src/Symfony/Component/HttpKernel/CacheClearer/Psr6CacheClearer.php

```diff
@@ -20,9 +20,27 @@ class Psr6CacheClearer implements CacheClearerInterface
 {
     private $pools = array();
 
+    public function __construct(array $pools = array())
+    {
+        $this->pools = $pools;
+    }
+
     public function addPool(CacheItemPoolInterface $pool)
     {
-        $this->pools[] = $pool;
+        @trigger_error(sprintf('The %s() method is deprecated since version 3.3 and will be removed in 4.0. Pass an array of pools indexed by name to the constructor instead.', __METHOD__), E_USER_DEPRECATED);
+
+        if (!in_array($pool, $this->pools, true)) {
```

Comment:
It's about keeping BC. If someone rely on `addPool()`, the pool should still be cleared when calling `clear()`, right?  The use of `in_array` is to avoid registering the same pool twice in this case.  Maybe you've a better one?